e2e/testcafe-devextreme/tests/dataGrid/common/keyboardNavigation/startEditing.functional.ts

```diff
@@ -36,6 +37,84 @@ test('Editing should start by pressing enter after scrolling content with scroll
   });
 });
 
+test('editing.allowUpdating callback should receive correct row on tab key on first cell with virtual scrolling (T1290811)', async (t) => {
+  const dataGrid = new DataGrid(DATA_GRID_SELECTOR);
+
+  await dataGrid.scrollBy({ y: 10000 });
+
+  await t
+    .click(dataGrid.getDataCell(49, 0).element)
+    .expect(dataGrid.getDataCell(49, 0).getEditor().element.focused).ok();
+
+  await t
+    .pressKey('tab')
+    .expect(dataGrid.getDataCell(49, 1).getEditor().element.focused).ok();
+
+  const eventRowKeys = await ClientFunction(() => (window as any).eventRowKeys)();
+
+  await t.expect(eventRowKeys).eql([49, 49, 49, 49, 49, 49]);
```

Comment:
Why was the allowUpdating callback called 6 times? The check is not obvious. If we don’t care about the number of times this callback is called, maybe it’s enough to check that there are no invalid values among the keys?
Or we could add only unique keys to the eventRowKeys array and check those.